lua/telescope/config.lua

```diff
@@ -190,6 +190,21 @@ local telescope_defaults = {
 
     path_display can also be set to 'hidden' string to hide file names
 
+    path_display can also be set to a function for custom formatting of the
+    path display. Example:
+
+        -- Format path as "file.txt (path\to\file\)"
+        path_display = function(opts, filepath)
+            iterator = string.gmatch(filepath, "(.-)([^\\/]-%.?([^%.\\/]*))$")
+            result = filepath
+            for key, value in iterator do
+                if key ~= '' then
+                    result = value .. " (" .. key .. ")"
+                end
```

Comment:
Let's make sure to review this example code as well. My Lua-fu is not strong yet. Seems that I need to make that check for '' since the match returns an extra empty value for some reason.